Merged _until and basic foreach methods.

etienneschmitt opened the pull request.

etienneschmitt committed:
Added Map3::boundary_face_of_Vertex(Vertex) const method.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap3  foreach_incident_cell_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap2 foreach_incident_cell_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Minor style updates.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap1 foreach_incident_cell_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
CMap2 : foreach_adjacent_xxx_through_yyy_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Map3 : foreach_adjacent_xxx_through_yyy_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge _until and the basic foreach methods.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merged foreach_cell and foreach_cell_until

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

sylvainthery commented:
:+1: 
On déplace juste un peu plus le curseur vers la simplicité d'écriture en l'éloignant un peu plus
de la lisibilité. Mais comme globalement on a déjà fait ce choix pour d'autres fonctions.

Et en fait le`foreach_bool_binder` n'a rien à voir avec foreach, il convertit si nécessaire une 
fonction void en bool qui renvoie true, je propose `default_true` ou `void_to_true`

pierrekraemer commented:
Oui moi aussi à propos du nom de la fonction foreach_bool_binder c'est le mot "foreach" qui m'embête le plus.

etienneschmitt commented:
void_to_true_binder ?

pierrekraemer commented:
pourquoi pas :-)

sylvainthery commented:
:+1:

etienneschmitt committed:
foreach_bool_binder --> void_to_true_binder

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed buf in MapBase::foreach_cell

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge foreach foreach_until for cmap3 tetra.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

pierrekraemer commented:
On est bon là ?

etienneschmitt committed:
Merge foreach foreach_until cmap3 hexa.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge foreach foreach_until cmap2 tri.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merged foreach foreach_until cmap2 quad.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Merge foreach_index and foreach_index_until in ChunkArrayContainer.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt committed:
Fixed some typos.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt commented:
Maintenant oui.

pierrekraemer commented:
Est-ce que les tests passent ou non dans l'état actuel ?

sylvainthery commented:
il semble qu'il y ait un seg. fault dans les test  avec Clang, mais difficile de voir ou dans les logs

etienneschmitt committed:
Fixed missing return.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>

etienneschmitt commented:
J'avais zappé un "return" dans cmap2_quad que je viens de corriger. ça devrait fonctionner ce coup-ci.

etienneschmitt commented:
Je pense qu'il me reste d'autres choses à modifier, il reste des endroits où je marque les cellules après l'execution de la lambda, alors que ça devrait être le contraire. Je corrige ça.

etienneschmitt committed:
foreach_ methods : when using markers, we now mark the orbit before calling the lambda function.
This is important when the lambda perform topological changes and add new darts that we don't want to mark.

Signed-off-by: Étienne Schmitt <etienne.schmitt@inria.fr>
